funnel/views/index.py

```diff
@@ -1,17 +1,19 @@
 # -*- coding: utf-8 -*-
 
 from flask import g, render_template, redirect, jsonify
+from coaster.views import jsonp, load_model
 from .. import app
 from ..models import Profile, ProposalSpace, Proposal
-from coaster.views import jsonp, load_model
 from .space import space_data
 
 
 @app.route('/')
 def index():
     g.profile = None
     g.permissions = []
-    spaces = ProposalSpace.query.filter_by(parent_space=None).filter(ProposalSpace.profile != None).filter(ProposalSpace.state.CURRENTLY_LISTED).order_by(ProposalSpace.date.desc()).all()  # NOQA
+    spaces = ProposalSpace.query.filter_by(parent_space=None).filter(
+        ProposalSpace.profile != None
+    ).filter(ProposalSpace.state.CURRENTLY_LISTED).order_by(ProposalSpace.date.desc()).all()  # NOQA
```

Comment:
We have to change the sorting from date descending to absolute number of days ±today, future events first. That means this query needs to be a union: future events with date ascending union with past events, date descending. Union queries: http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.union

Reply:
This also applies to other ProposalSpace queries below.